src/charm.py

```diff
@@ -187,22 +249,25 @@ def configure_workload_container(self, event) -> None:
                 logger.info(msg)
                 self.unit.status = MaintenanceStatus(msg)
                 container.add_layer(CONTAINER_NAME, pebble_config, combine=True)
+                container.pebble.replan_services()
+            else:
```

Comment:
Why this else? It is my understanding that `if services != pebble_config["services"]` will check if they are changes, if not, we can just skip the snippet enclosed in the `if`